distributed/benchmarks/client.py

```diff
@@ -37,3 +33,38 @@ def inc(x):
         total = delayed(sum, pure=True)(L)
 
         total.compute(scheduler=self.client)
+
+
+class WorkerRestrictionsSuite(object):
+
+    def setup(self):
+        cluster = LocalCluster(n_workers=1, threads_per_worker=1,
+                               resources={"resource": 1}, worker_class=Worker)
+        spec = copy.deepcopy(cluster.new_worker_spec())
+        del spec[1]['options']['resources']
+        cluster.worker_spec.update(spec)
+        cluster.scale(2)
+        client = Client(cluster)
+
+        self.client = client
+
+    def teardown(self):
+        self.client.close()
+
+    def time_trivial_tasks_restrictions(self):
+        """
+        Benchmark measuring the improvment from allowing new workers
+        to steal tasks with resource restrictions.
+        """
+        client = self.client
+
+        info = client.scheduler_info()
+        workers = list(info['workers'])
+        futures = client.map(slowinc, range(10),
+                             delay=0.1, resources={"resource": 1})
+        client.cluster.scale(len(workers) + 1)
```

Comment:
The 100ms cost here is likely to dwarf the task overhead that we're looking to find here, which is likely to be in the 10us range.